src/h.ts

```diff
@@ -17,29 +17,48 @@ function addNS(data: any, children: VNodes | undefined, sel: string | undefined)
   }
 }
 
+function flattenChildren(children: VNodeChildren[], flattened: VNode[]) {
+  for (const child of children) {
+    if (child !== undefined && child !== null && child !== false) {
+      if (is.array(child)) {
+        flattenChildren(child, flattened);
+      } else if (is.primitive(child)) {
+        flattened.push(vnode(undefined, undefined, undefined, String(child), undefined));
+      } else {
+        flattened.push(child);
+      }
+    }
+  }
+  return flattened;
+}
+
 export function h(sel: string): VNode;
 export function h(sel: string, data: VNodeData): VNode;
 export function h(sel: string, children: VNodeChildren): VNode;
-export function h(sel: string, data: VNodeData, children: VNodeChildren): VNode;
+// jsx/tsx compatible h signature
+export function h(sel: string | Function, data: VNodeData | null, ...children: VNodeChildren[]): VNode;
 export function h(sel: any, b?: any, c?: any): VNode {
-  var data: VNodeData = {}, children: any, text: any, i: number;
+  var data: VNodeData = {}, children: any, text: any;
   if (c !== undefined) {
-    data = b;
+    if (arguments.length > 3) {
+      c = Array.prototype.slice.call(arguments, 2);
+    }
+    if (b !== null) data = b;
     if (is.array(c)) { children = c; }
     else if (is.primitive(c)) { text = c; }
     else if (c && c.sel) { children = [c]; }
-  } else if (b !== undefined) {
+  } else if (b !== undefined && b !== null) {
     if (is.array(b)) { children = b; }
     else if (is.primitive(b)) { text = b; }
     else if (b && b.sel) { children = [b]; }
     else { data = b; }
   }
   if (children !== undefined) {
-    for (i = 0; i < children.length; ++i) {
-      if (is.primitive(children[i])) children[i] = vnode(undefined, undefined, undefined, children[i], undefined);
-    }
+    children = flattenChildren(children, []);
   }
-  if (
+  if (typeof sel === `function`) {
+    return sel(data, children);
```

Comment:
When is this the case, please?

Reply:
This is when a FunctionComponent is used. I've added a better type for that e.g 

```ts
import {h, init} from 'snabbdom';
const patch = init([..]);

function App() {
  const greeting = 'Hello Function Component!';
  return <Headline value={greeting} />;
}

function Headline({value}) {
  return <h1>{value}</h1>;
}

patch(document.getElementById(`app`), <App />)
```

jsx treats any functions begining with capital letter e.g `<Headline>` as components and small letters e.g `<h1>` as intrinsic elements. A function component is basically a function that takes some props and returns vnodes. This allows composition of a large h call tree into smaller re-usable functions. Something very encouraged when building large views. In h world they get transpiled to `h(Headline, {value: 'Hello Function Component'})`. So we're essentially just calling the function to retrieve the children equivalent to `Headline({value: 'Hello Function Component'})`